frame/staking/src/pallet/mod.rs

```diff
@@ -225,7 +225,10 @@ pub mod pallet {
 		/// staker. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
 		///
 		/// Invariant: what comes out of this list will always be a nominator.
-		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
+		///
+		/// NOTE: Staking does not maintain this list, it merely reads from it. The list is
+		/// maintained by `EventListener` implementors.
```

Comment:
I think these are all assumptions that are not part of the API, but are instead "implementation notes". Nowhere in any of these interfaces it is hardcoded that `VoterList` MUST be implemented by whatever is behind `EventListeners`. In fact, we have `UseNominatorsMap` which contradicts this. 


```suggestion
		// NOTE: Staking does not maintain this list, it merely reads from it. The list is
		// maintained by `EventListener` implementors.
```

Reply:
I am happy to keep some degree of such notes around the code if you want, but you can always document implementation details in the PR description, and thanks to tools like git blame you can always re-discover them.

Reply:
I think you are right, at the same time having hints in the codebase as opposed to having to use `git blame`, which eventually might be overwritten, saves a lot of time. It's not hardcoded, but it's the current design, when the design changes - we can always change the comments. My goal here is to deliver info in the more straightforward way possible, but I definitely don't want to push this, if it does not seem to line up with the rest of the codebase.

Reply:
It's the comments like this that make a difference for a builder: whether they have to go to stackExchange to ask a question or have an easy suggestion to keep digging. At least from my perspective of a former builder.

Reply:
This is an outdated diff, already addressed a while ago.